app/services/reference_number_service.rb

```diff
@@ -0,0 +1,17 @@
+class ReferenceNumberService
+  RANDOM_SOURCE_SET_1 = %w[A B C D E F G H J K L M N P R S T U V W X Y].freeze
+  RANDOM_SOURCE_SET_2 = %w[2 3 4 5 6 7 8 9].freeze
+  RANDOM_SOURCE_UNION = RANDOM_SOURCE_SET_1 + RANDOM_SOURCE_SET_2
```

Comment:
Optional, but I think it might be easier to see what this does by giving these more specific names:

```suggestion
  ALPHABETICAL_CHARACTER_SET = %w[A B C D E F G H J K L M N P R S T U V W X Y].freeze
  NUMERIC_CHARACTER_SET = %w[2 3 4 5 6 7 8 9].freeze
  ALPHANUMERIC_CHARACTER_SET = ALPHABETICAL_CHARACTER_SET + NUMERIC_CHARACTER_SET
```

Reply:
Sorted, good suggestion!